toolkit/authentication/src/test/java/com/arcgismaps/toolkit/authentication/SafeSuspendableCancellableTest.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ *
+ *  Copyright 2025 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.arcgismaps.toolkit.authentication
+
+import com.arcgismaps.toolkit.authentication.utils.safeSuspendableCancellable
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.async
+import kotlinx.coroutines.test.runTest
+import org.junit.Ignore
+import org.junit.Test
+import java.util.concurrent.atomic.AtomicBoolean
+
+class SafeSuspendableCancellableTest {
+
+    /**
+     * Given a safeSuspendableCancellable block
+     * When the block resumes successfully
+     * Then the value is returned
+     * @since 200.8.1
+     */
+    @Test
+    fun returnsValueOnSuccess() = runTest {
+
+        val result = safeSuspendableCancellable<String> { cont ->
+            cont.resumeSafely("value")
+        }
+        assertThat(result).isEqualTo("value")
+    }
+
+    /**
+     * Given a safeSuspendableCancellable block
+     * When the block throws an exception
+     * Then the exception is propagated
+     * @since 200.8.1
+     */
+    @Test
+    fun throwsOnFailure() = runTest {
+        val ex = RuntimeException("boom")
+        var caught: Throwable? = null
+        try {
+            safeSuspendableCancellable<Unit> { _ ->
+                throw ex
+            }
+        } catch (t: Throwable) {
+            caught = t
+        }
+        assertThat(caught).isInstanceOf(ex::class.java)
+        assertThat(caught?.message).isEqualTo(ex.message)
+    }
+
+    /**
+     * Given a safeSuspendableCancellable block
+     * When the block attempts to resume after throwing an exception
+     * Then the resume is ignored and the exception is propagated
+     * @since 200.8.1
+     */
+    @Test
+    fun throwExceptionFirstThenReturnValueIsIgnored() = runTest {
+        var safeResult: String? = null
+        val res = runCatching {
+            safeResult = safeSuspendableCancellable<String> { cont ->
+                throw IllegalStateException("This should be called first")
+                cont.resumeSafely("late") // ignored
```

Comment:
Needs a suppress annotation:
```suggestion
                @Suppress("UNREACHABLE_CODE")
                cont.resumeSafely("late") // ignored
```

Reply:
Edit: I think this test is redundant. `throwsOnFailure` is already testing what you are testing here

Reply:
removed test